### PR TITLE
Edit teams in settings window

### PR DESCRIPTION
**Pull request by MetalCar**
The users can now edit their configured teams in settings window. I only tested it on windows. I also did changes on the UI for adding teams, because I think on this way it's much clearer how to add a team and how to edit a team. This related to #3.

`npm test`ran successfully except this `prettify`. I don't know if I got it right, but I called the command `npm run prettify`and I doesn't get an error there.

Cheers

### CODE REVIEW

**Comment by MetalCar** [comment]
Found that `prettify` thing by myself. Did some weird jsx-code there.

**Comment by MetalCar** [comment]
I forgot to put an image there how my changes looks like.
editteamsettings

**Comment by MetalCar** [comment]
I found out that when I first edit a team and them add a new one, that I still edit the last selected team. The problem were the states. Hope that I get it right now.

**Comment by yuya-oc** [comment]
Thanks! Just curious, is it possible to move '+' button to near the 'Team' label? For example, right of the label. The button is far from the label, so I think the user can't notice the button when adding a new team.

@CarmDam Could you complete Mattermost CLA?

**Comment by MetalCar** [comment]
yes i will move the button next to the label when i'm back home.

I did the mattermost cla as MetalCar @CarmDam is also my account. it seems that i pushed my last commit with the wrong github account

**Comment by MetalCar** [comment]
@yuya-oc Is this OK? Or should I place the button nearer to the `Teams`-Headline? 
image

**Comment by yuya-oc** [comment]
Looks better. Could you try resizing into narrow width?

**Comment by MetalCar** [comment]
Do you mean resize the complete window? Like that?
image

**Comment by yuya-oc** [comment]
Sorry for few description, more narrow. I want to know whether Bootstrap correctly layouts the button in too narrow window (like mobile devices).

**Comment by MetalCar** [comment]
Ah, now I understand what you want. Hope when i push my changes, that all is OK....first time I used bootstrap with reactJS
image

**Comment by yuya-oc** [comment]
Thanks. That looks OK!
Please let me confirm, you haven't pushed the change yet, right?

**Comment by MetalCar** [comment]
Yep, i will push it now.

File: src/browser/settings.jsx [hunk 1]
```diff
@@ -28,14 +28,21 @@ var SettingsPage = React.createClass({
     } catch (e) {
       config = settings.loadDefault();
     }
+
+    this.setState({
```
Comment (Razzeee) [inline src/browser/settings.jsx]: This causes an warning in the console log.
```
settings.js:982 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the SettingsPage component.warning @ settings.js:982
```
Removing this code, solves it, not sure if it has any drawbacks/problems/how to do it properly.

Comment (Razzeee) [inline src/browser/settings.jsx]: This might be as simple as ```config.showAddTeamForm = false;``` instead of this.

Comment (yuya-oc) [inline src/browser/settings.jsx]: Should be `config.showAddTeamForm = false;`. As of warning in the console log, `setState()` can't be used in `getInitialState()` of React.js.

**Comment by magicmonty** [comment]
This seems not to work correctly on OS X

screenflow

**Comment by yuya-oc** [comment]
Does [this build](https://circleci.com/gh/mattermost/desktop/456#artifacts) fix problems?

**Comment by magicmonty** [comment]
I have tested this build on OSX, and the problem is gone

**Comment by yuya-oc** [comment]
Thanks, I will merge later.

**Comment by jasonblais** [comment]
Thanks @magicmonty, @yuya-oc!!

**Pull request merged**